### PR TITLE
PLFM-6527: Build stack from snapshots

### PR DESCRIPTION
**Pull request by xschildw**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImpl.java [hunk 1]
```diff
@@ -317,24 +319,40 @@ VelocityContext createSharedContext() {
...
 				.withMaxAllocatedStorage(config.getIntegerProperty(PROPERTY_KEY_REPO_RDS_MAX_ALLOCATED_STORAGE))
 				.withInstanceIdentifier(stack + "-" + instance + "-db").withDbName(stack + instance)
 				.withInstanceClass(config.getProperty(PROPERTY_KEY_REPO_RDS_INSTANCE_CLASS))
 				.withDbStorageType(config.getProperty(PROPERTY_KEY_REPO_RDS_STORAGE_TYPE))
 				.withDbIops(config.getIntegerProperty(PROPERTY_KEY_REPO_RDS_IOPS))
 				.withMultiAZ(config.getBooleanProperty(PROPERTY_KEY_REPO_RDS_MULTI_AZ));
+		String repoSnapshotIdentifier = config.getProperty(PROPERTY_KEY_RDS_REPO_SNAPSHOT_IDENTIFIER);
+		if (! NOSNAPSHOT.equals(repoSnapshotIdentifier)) {
+			results[0] = repoDbDescriptor.withSnapshotIdentifier(repoSnapshotIdentifier);
+		} else {
```
Comment (marcomarasca) [inline src/main/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImpl.java]: We can remove the else as we always need the assignement

File: src/main/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImpl.java [hunk 2]
```diff
@@ -317,24 +319,40 @@ VelocityContext createSharedContext() {
...
+			results[0] = repoDbDescriptor;
+		}
 
+		String[] repoTableSnapshotIdentifiers = null;
+		if (repoSnapshotIdentifier != null) {
```
Comment (marcomarasca) [inline src/main/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImpl.java]: Maybe better to throw an exception instead of a "hidden" default

File: src/test/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImplTest.java [hunk 3]
```diff
@@ -332,8 +341,60 @@ public void testBuildAndDeployDev() throws InterruptedException {
...
+		assertFalse(resources.has("dev101Table1RepositoryDBAlarmSwapUsage"));
+		assertFalse(resources.has("devd101Table1RepositoryDBHighWriteLatency"));
+		assertFalse(resources.has("dev101Table1RepositoryDBHighCPUUtilization"));
+		assertFalse(resources.has("dev101Table1RepositoryDBLowFreeStorageSpace"));
+
+		assertTrue(resources.has("dev101RepositoryDB"));
+		JSONObject repoDB = (JSONObject)resources.get("dev101RepositoryDB");
+		JSONObject dbProps = (JSONObject)repoDB.get("Properties");
+		assertTrue(dbProps.has("DBSnapshotIdentifier"));
+		assertFalse(dbProps.has("DBName"));
```
Comment (marcomarasca) [inline src/test/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImplTest.java:431]: We should probably check the tables identifiers as well and the edge case where there is an inconsistency between the main and tables identifiers "presence"
